src/main/java/seedu/address/logic/commands/DeleteElderlyCommand.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Elderly;
+import seedu.address.model.person.NricMatchesKeywordsPredicate;
+
+/**
+ * Deletes an elderly identified using its NRIC from the FriendlyLink database.
+ */
+public class DeleteElderlyCommand extends Command {
+    public static final String COMMAND_WORD = "delete_elderly";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the elderly identified by the nric linked with the elderly.\n"
+            + "Parameters: NRIC (must be a valid NRIC number)\n"
+            + "Example: " + COMMAND_WORD + " S1234567C";
+
+    public static final String MESSAGE_DELETE_ELDERLY_SUCCESS = "Deleted Elderly: %1$s";
+
+    private final String targetNric;
+
+    public DeleteElderlyCommand(String targetNric) {
+        this.targetNric = targetNric;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        // first filter with NRIC, then get filtered list
+        NricMatchesKeywordsPredicate predicate =
+                new NricMatchesKeywordsPredicate(targetNric);
+        model.updateFilteredElderlyList(predicate);
```

Comment:
The logic here doesn't seem right. This `filteredList` is the list that will be displayed on the GUI. By calling `model.updateFilteredElderlyList(predicate)`, you're effectively updating the UI to only show the current elderly that is being deleted. Take a look at `MainWindow.java`, line 117. Instead, what we can do for this PR is to map the corresponding elderly index to the elderly NRIC. If it is too much trouble, a trivial way would be to do an O(n) loop for now and search it through the list. 

Do write a `// TODO:` for future fixes.

Reply:
Oh ya, do remove the relevant parts from `DeleteCommand` in your PR!

Reply:
We currently do not need to touch the ```filteredList``` since its just a wrapper around the internal list to have extra predicate for filtering. When you call ```removeElderly()``` it  removes the elderly from the the internal list and  ```filteredList``` will also display accordingly

Reply:
Then what do you feel about the idea of adding a `getFilteredElderlyList(predicate)` in model?